React/CoreModules/RCTDeviceInfo.mm

```diff
@@ -202,16 +207,16 @@ - (void)interfaceFrameDidChange
 - (void)_interfaceFrameDidChange
 {
   NSDictionary *nextInterfaceDimensions = RCTExportedDimensions(_moduleRegistry, _bridge);
-
-  if (!([nextInterfaceDimensions isEqual:_currentInterfaceDimensions])) {
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated-declarations"
-    [[_moduleRegistry moduleForName:"EventDispatcher"] sendDeviceEventWithName:@"didUpdateDimensions"
-                                                                          body:nextInterfaceDimensions];
-#pragma clang diagnostic pop
+    UIApplicationState appState = [RCTSharedApplication() applicationState];
+    
+  if (!([nextInterfaceDimensions isEqual:_currentInterfaceDimensions]) && appState == UIApplicationStateActive) {
+    #pragma clang diagnostic push
+    #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+        [[_moduleRegistry moduleForName:"EventDispatcher"] sendDeviceEventWithName:@"didUpdateDimensions"
+                                                                              body:nextInterfaceDimensions];
+          _currentInterfaceDimensions = nextInterfaceDimensions;
+    #pragma clang diagnostic pop
   }
-
-  _currentInterfaceDimensions = nextInterfaceDimensions;
```

Comment:
I added a comment in code, but that's moved inside so that we make sure that the state between JS and Obj-C is synced. Meaning we only update the 'Native' state when it's published to JS.